src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,18 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return numbers.stream()
+                .map(number -> {
+                    if (isOdd) {
+                        number--;
+                    }
+                    isOdd = !isOdd;
+                    return number;
+                })
```

Comment:
```suggestion
                .map(number -> {
                    if (isOdd) {
                        number--;
                    }
                    isOdd = !isOdd;
                    return number;
                })
```I used counter here, your solution looks better than mine.